lib/ansible/module_utils/aws/urls.py

```diff
@@ -38,22 +39,46 @@ def sign(key, msg):
     '''
     Return digest for key applied to msg
     '''
+
     return hmac.new(key, msg.encode('utf-8'), hashlib.sha256).digest()
 
 
 def get_signature_key(key, dateStamp, regionName, serviceName):
     '''
     Returns signature key for AWS resource
     '''
+
     kDate = sign(('AWS4' + key).encode('utf-8'), dateStamp)
     kRegion = sign(kDate, regionName)
     kService = sign(kRegion, serviceName)
     kSigning = sign(kService, 'aws4_request')
     return kSigning
 
 
+def get_aws_key_pair(module):
+    '''
+    Returns aws_access_key_id, aws_secret_access_key for a module.
+
+    Prefers the credentials from a profile over inline'd keys
+    '''
+
+    if not HAS_BOTO3:
+        module.fail_json("get_aws_key_pair requires boto3")
+
+    dummy, dummy, boto_params = get_aws_connection_info(module, boto3=True)
+
+    profile = boto_params.get('profile_name')
+
+    if profile:
+        s = session.Session(profile_name=profile)
+        credentials = s.get_credentials()
+        return credentials.access_key, credentials.secret_key
```

Comment:
What about the session token? (That this stuff is hard to do is why `connect_to_aws` was abstracted so early on!)